library/src/updater.rs

```diff
@@ -290,43 +324,28 @@ fn update_internal(_: &UpdaterLockState) -> anyhow::Result<UpdateStatus> {
     // Saves state to disk (holds Config lock while writing).
 
     let config = copy_update_config()?;
-    // We should never try to write this state as some other writer may be
-    // racing with us, we should get a new state inside a lock if we want
-    // to write.
-    let read_only_state = UpdaterState::load_or_new_on_error(
-        &config.storage_dir,
-        &config.release_version,
-        config.patch_public_key.as_deref(),
-    );
 
     // We discard any events if we have more than 3 queued to make sure
     // we don't stall the client.
-    let events = read_only_state.copy_events(3);
+    let events = with_state(|state| Ok(state.copy_events(3)))?;
     for event in events {
         let result = crate::network::send_patch_event(event, &config);
         if let Err(err) = result {
             error!("Failed to report event: {:?}", err);
         }
     }
-    // We're abusing the config lock as a UpdateState lock for now.
-    let read_only_state = with_config(|_| {
```

Comment:
This is what I'm referring to when I say that we're no longer accessing an `UpdaterState` outside of a lock.